R/gh_response.R

```diff
@@ -1,33 +1,35 @@
-gh_process_response <- function(response) {
-  stopifnot(inherits(response, "response"))
-  if (status_code(response) >= 300) {
-    gh_error(response)
+gh_process_response <- function(resp) {
+  stopifnot(inherits(resp, "httr2_response"))
+  if (httr2::resp_status(resp) >= 300) {
+    gh_error(resp)
   }
 
-  content_type <- http_type(response)
-  gh_media_type <- headers(response)[["x-github-media-type"]]
-  is_raw <- content_type == "application/octet-stream" ||
+  content_type <- httr2::resp_content_type(resp)
+  gh_media_type <- httr2::resp_header(resp, "x-github-media-type")
+
+  is_raw <- identical(content_type, "application/octet-stream") ||
     isTRUE(grepl("param=raw$", gh_media_type, ignore.case = TRUE))
-  is_ondisk <- inherits(response$content, "path")
+  is_ondisk <- inherits(resp$body, "httr2_path")
+
   if (is_ondisk) {
-    res <- response$content
+    res <- as.character(resp$body)
+  } else if (is.na(content_type)) {
+    # empty body
+    res <- list()
   } else if (grepl("^application/json", content_type, ignore.case = TRUE)) {
-    res <- fromJSON(content(response, as = "text"), simplifyVector = FALSE)
+    res <- httr2::resp_body_json(resp)
   } else if (is_raw) {
-    res <- content(response, as = "raw")
-  } else if (content_type == "application/octet-stream" &&
-    length(content(response, as = "raw")) == 0) {
-    res <- NULL
+    res <- httr2::resp_body_raw(resp)
   } else {
     if (grepl("^text/html", content_type, ignore.case = TRUE)) {
       warning("Response came back as html :(", call. = FALSE)
     }
-    res <- list(message = content(response, as = "text"))
+    res <- list(message = httr2::resp_body_string(resp))
   }
 
-  attr(res, "method") <- response$request$method
-  attr(res, "response") <- headers(response)
-  attr(res, ".send_headers") <- response$request$headers
+  # attr(res, "method") <-  response$request$method
```

Comment:
I can bring these back if you think they're important, but nothing tests them currently.

Reply:
I agree that probably only the response headers are important. OTOH, we can keep things as they were, just to be on the safe side.

Reply:
Slightly hacky, but now restored.